src/pages/Typing/Typing.tsx

```diff
@@ -0,0 +1,135 @@
+import { Fragment, FC, useState } from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+import { Container, Paper, Grid, TextField } from '@material-ui/core'
+import { TypingHeader, Typography, Result, Loader } from '../../components'
+import { typingStyles } from './Typing.styles'
+import { saveTestResult } from '../../redux'
+
+interface TypingProps {}
+
+const Typing: FC<TypingProps> = () => {
+  const {
+    testData: { testInfo: { paragraph = '', level = '', time = 0 } = {} } = {},
+    userLogin: { userInfo: { displayName = '', photoURL = '' } = {} } = {},
+  } = useSelector((state: any) => state)
+  // const [wordsArray, setWordsArray] = useState<any>([])
+  const dispatch = useDispatch()
+  const classes = typingStyles()
+
+  const [correctEntries, setCorrectEntries] = useState<number>(0)
+  const [inCorrectEntries, setInCorrectEntries] = useState<number>(0)
+  const [greenArray, setGreenArray] = useState<Array<number>>([])
+  const [redArray, setRedArray] = useState<Array<number>>([])
+  const [stop, setStop] = useState<boolean>(false)
+  const [result, setResult] = useState({
+    accuracy: 0,
+    netWPM: 0,
+  })
+
+  // useEffect(() => {
+  //   const wordArr = paragraph.split(' ')
+  //   setWordsArray(wordArr)
+  // }, [paragraph])
+  const handleText = ({ target: { value = '' } = {} }) => {
+    value === paragraph && handleTimeStop()
+    //while typing inserting each letter that is correct into green array and wrong one in red
+    if (value.charAt(value.length - 1) === paragraph.charAt(value.length - 1)) {
+      setCorrectEntries(correctEntries + 1)
+      setGreenArray([...greenArray, value.length - 1])
+    } else {
+      setInCorrectEntries(inCorrectEntries + 1)
+      setRedArray([...redArray, value.length - 1])
+    }
+    // removing letter from input box when user click backspace
+    if (greenArray[greenArray.length - 1] > value.length - 1) {
+      setGreenArray(greenArray.filter((item) => item !== value.length))
+      setCorrectEntries(correctEntries - 1)
+    } else if (redArray[redArray.length - 1] > value.length - 1) {
+      setInCorrectEntries(inCorrectEntries - 1)
+      setRedArray(redArray.filter((item) => item !== value.length))
+    }
+  }
+
+  const handleTimeStop = () => {
+    setStop(true)
+    const timeSpent =
+      // @ts-ignore
```

Comment:
Better remove this ignore and handle the case when `localStorage.getItem('time-remaining')` can be null gracefully?